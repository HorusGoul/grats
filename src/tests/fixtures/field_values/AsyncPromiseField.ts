/** @gqlType */
export default class SomeType {
  /** @gqlField */
  async hello(): Promise<string> {
    return "Hello world!";
  }
}
