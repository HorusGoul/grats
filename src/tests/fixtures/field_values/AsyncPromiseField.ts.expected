-----------------
INPUT
----------------- 
/** @gqlType */
export default class SomeType {
  /** @gqlField */
  async hello(): Promise<string> {
    return "Hello world!";
  }
}

-----------------
OUTPUT
-----------------
directive @asyncIterable on FIELD_DEFINITION

directive @propertyName(name: String!) on FIELD_DEFINITION

directive @exported(tsModulePath: String!, functionName: String!, argCount: Int!) on FIELD_DEFINITION

type SomeType {
  hello: String
}